### DISCUSSION TITLE
What is this project precisely? - I have many questions listed. + Cannot run "py webui.py" -> [Errno 2] No such file or directory: 'Data/config.json'

### DISCUSSION THREAD

**Discussion by APISeeker** [fishaudio/Bert-VITS2#282] [Q&A]
Hello,
I made 3 parts:
Part1:
I think I am getting confused since the languages are not always in english (and I respect that), I just want to know if I am not missing on some important technology or not.
https://imgur.com/t9ZepOF

So Here are my questions to clear things up and hopefully everythin will be clear:

What is this technolgy about? Is it similar to rvs tts edge?
what are its limitation? Can I reproduce "text" to "voice" with it? In any "tone"?
What is difference between tone and emotion? (Emotion seems to be in another repo, but I still want to understand please)
What is the difference between this and the new repository mentioned in the "last release" description (FISH SPEECH)?
Thanks so much ^^

_____________
Part2:
Emotions and tone are 2 distinct things right? Sorry the UI was in japanese and I only found this yesterday I did not have time to understand everything.

To install this repo (yours), I just need to to go to "releases", take the zip content, then do pip install -r requirements.txt, right?
I saw that you guys wrote a commet about going to another repository (fish speech I think?)
-> Can you tell me what is the difference betwene the 2 and what each can do better than the other? Thanks again and sorry for so many questions, I will try to understand everything in ht next 48H but some input from you might help me speed up the process of understanding what's going on ^^

_______
Part3:
```
py webui.py
已根据默认配置文件default_config.yml生成配置文件config.yml。请按该配置文件的说明进行配置后重新运行 。
如无特殊需求，请勿修改default_config.yml或备份该文件。
pytorch_model.bin: 100%|██████████████████████████████████████| 1.32G/1.32G [00:10<00:00, 120MB/s]
pytorch_model.bin: 100%|██████████████████████████████████████| 1.31G/1.31G [00:11<00:00, 118MB/s]
spm.model: 100%|██████████████████████████████████████████████| 2.46M/2.46M [00:00<00:00, 123MB/s]
pytorch_model.bin: 100%|████████████████████████████████████████| 874M/874M [00:07<00:00, 121MB/s]
Traceback (most recent call last):
  File "C:\Users\aa\Downloads\Bert-VITS2-fix-4\webui.py", line 377, in <module>
    hps = utils.get_hparams_from_file(config.webui_config.config_path)
  File "C:\Users\aa\Downloads\Bert-VITS2-fix-4\utils.py", line 333, in get_hparams_from_file
    with open(config_path, "r", encoding="utf-8") as f:
FileNotFoundError: [Errno 2] No such file or directory: 'Data/config.json'
```

I think it can be fixed if we can change the value of the path to : 'configs/config.json' np?
![image](https://github.com/fishaudio/Bert-VITS2/assets/156720940/6da39b0e-4586-4428-bf90-ee4a2b265721)

But I don't know which file contain the info Data/config.json, so I can modify it.

**Comment by karry12138** [comment]
Bro, I've met the same problem in Part3, and I've solved this. (struggling for 1 hour).
In "default_config.yml",focus on the top few lines:
```
# 每个数据集与其对应的模型存放至统一路径下，后续所有的路径配置均为相对于datasetPath的路径
# 不填或者填空则路径为相对于项目根目录的路径
dataset_path: "Data/"
```
To regulate the path of dataset and model, all of the path below this line will be relative to [dataset_parh],which means "Data/".That explains the error
```
"[Errno 2] No such file or directory: 'Data/config.json'"
```

By the way,the config.json you've mentioned (configs/config.json) is just the one what it needs.
So you can solve this by creating a directory "Data" in the root and copy the "configs/config.json" to "Data/config.json".
Don't forget to change other path in the same way(if needed)

**Comment by APISeeker** [comment]
Thanks @karry12138 , It worked but I have another error now.
Note:
Funny enough I notice there is a DATA file in oen of the other releases such as "Bert-VITS2-2.3-Manager230", do you inderstand what is the difference between all these releases?

The new error:
```
C:\Users\aaa\Downloads\Bert-VITS2-fix-4\venvBERTVITS2\lib\site-packages\torch\nn\utils\weight_norm.py:30: UserWarning: torch.nn.utils.weight_norm is deprecated in favor of torch.nn.utils.parametrizations.weight_norm.
  warnings.warn("torch.nn.utils.weight_norm is deprecated in favor of torch.nn.utils.parametrizations.weight_norm.")
Traceback (most recent call last):
  File "C:\Users\aaa\Downloads\Bert-VITS2-fix-4\webui.py", line 380, in <module>
    net_g = get_net_g(
  File "C:\Users\aaa\Downloads\Bert-VITS2-fix-4\infer.py", line 98, in get_net_g
    _ = utils.load_checkpoint(model_path, net_g, None, skip_optimizer=True)
  File "C:\Users\aaa\Downloads\Bert-VITS2-fix-4\utils.py", line 66, in load_checkpoint
    assert os.path.isfile(checkpoint_path)
AssertionError
```
